erpnext/selling/doctype/sales_order/tests/test_sales_order_with_bypass_credit_limit_check.js

```diff
@@ -0,0 +1,54 @@
+QUnit.module('Sales Order');
+
+QUnit.test("test_sales_order_with_bypass_credit_limit_check", function(assert) {
+//#PR : 10861, Author : ashish-greycube & jigneshpshah,  Email:mr.ashish.shah@gmail.com 
+	assert.expect(2);
+	let done = assert.async();
+	frappe.run_serially([
+		() => frappe.new_doc('Customer'),
+		() => frappe.timeout(1),
+		() => frappe.click_link('Edit in full page'),
+		() => cur_frm.set_value("customer_name", "Test Customer 10"),
+		() => cur_frm.set_value("credit_limit", 100.00),
+		() => cur_frm.set_value("bypass_credit_limit_check_at_sales_order", 1),
+		// save form
+		() => cur_frm.save(),
+		() => frappe.timeout(1),
+
+		() => frappe.new_doc('Item'),
+		() => frappe.timeout(1),
+		() => frappe.click_link('Edit in full page'),
+		() => cur_frm.set_value("item_code", "Test Product 10"),
+		() => cur_frm.set_value("item_group", "Products"),
+		() => cur_frm.set_value("standard_rate", 100),	
+		// save form
+		() => cur_frm.save(),
+		() => frappe.timeout(1),		
+
+		() => {
+			return frappe.tests.make('Sales Order', [
+				{customer: 'Test Customer 5'},
+				{items: [
+					[
+						{'delivery_date': frappe.datetime.add_days(frappe.defaults.get_default("year_end_date"), 1)},
```

Comment:
can you write these tests on the server side too?

Reply:
Rushabh, I did give a try for server side i.e. test_sales_order.py but the issue is it has make_sales_order function which doesn't use submit logic of sales_order.py. Server side does submit via frappe.new_doc and submit directly to DB. 
sales_order.py --> bypass_credit_limit_check_at_sales_order logic is in check_credit_limit function called before click of submit button.
Hence i have written the test case using QUnit. thanks